### PR TITLE
Use Props for for device permission dropdown

### PR DESCRIPTION
**Pull request by carocao-msft**
# What
Use Props for for device permission dropdown

# Why
https://skype.visualstudio.com/SPOOL/_workitems/edit/3002586

# How Tested
comment out askdevicepermission code in call composite 

https://user-images.githubusercontent.com/96077406/194130960-853b985e-1880-4a89-b489-ab3baf1a3f96.mov

### CODE REVIEW

File: packages/communication-react/review/beta/communication-react.api.md [hunk 1]
```diff
@@ -56,7 +56,7 @@ import { MediaStreamType } from '@azure/communication-calling';
 import { MicrosoftTeamsUserKind } from '@azure/communication-common';
 import type { NetworkDiagnosticChangedEventArgs } from '@azure/communication-calling';
 import { PartialTheme } from '@fluentui/react';
-import { PermissionConstraints } from '@azure/communication-calling';
+import { PermissionConstraints as PermissionConstraints_2 } from '@azure/communication-calling';
```
Comment (dmceachernmsft) [inline packages/communication-react/review/beta/communication-react.api.md:59]: I know we maybe have a precedent here to use a number but could we call it something else more specific to this dropdown? or callReadiness? the naming collision is with Rooms right?

File: packages/communication-react/review/beta/communication-react.api.md [hunk 2]
```diff
@@ -2593,6 +2595,14 @@ export type ParticipantsRemovedListener = (event: {
 // @public
 export type ParticipantState = 'Idle' | 'Connecting' | 'Ringing' | 'Connected' | 'Hold' | 'InLobby' | 'EarlyMedia' | 'Disconnected';
 
+// Warning: (ae-internal-missing-underscore) The name "PermissionConstraints" should be prefixed with an underscore because the declaration is marked as @internal
+//
+// @internal
+export type PermissionConstraints = {
```
Comment (dmceachernmsft) [inline packages/communication-react/review/beta/communication-react.api.md:2601]: Fix naming warning here

File: packages/communication-react/review/beta/communication-react.api.md (hunk 1 above)
Comment (edwardlee-msft) [inline packages/communication-react/review/beta/communication-react.api.md:59]: Interesting.. this is being imported as PermissionConstraints_2

Comment (carocao-msft) [reply packages/communication-react/review/beta/communication-react.api.md:59]: Rush build did this automatically, super weird...

**Review by dmceachernmsft — APPROVED** [review]

File: packages/react-components/review/stable/react-components.api.md [hunk 3]
```diff
@@ -1125,6 +1126,14 @@ export interface ParticipantsButtonStyles extends ControlBarButtonStyles {
 // @public
 export type ParticipantState = 'Idle' | 'Connecting' | 'Ringing' | 'Connected' | 'Hold' | 'InLobby' | 'EarlyMedia' | 'Disconnected';
 
+// Warning: (ae-internal-missing-underscore) The name "PermissionConstraints" should be prefixed with an underscore because the declaration is marked as @internal
+//
+// @internal
+export type PermissionConstraints = {
```
Comment (edwardlee-msft) [inline packages/react-components/review/stable/react-components.api.md:1132]: api warning generated, possibly need to rename with underscore?

File: packages/react-composites/src/composites/CallComposite/CallComposite.tsx [hunk 4]
```diff
@@ -329,34 +329,12 @@ export const CallComposite = (props: CallCompositeProps): JSX.Element => {
         adapter.querySpeakers();
         return;
       }
-      /* @conditional-compile-remove(call-readiness) */
```
Comment (edwardlee-msft) [inline packages/react-composites/src/composites/CallComposite/CallComposite.tsx:332]: Looks good.
Thanks for cleaning this up. Now with the buttons, it should be good to remove this.

**Review by edwardlee-msft — APPROVED** [review]

**Pull request merged**